includes/class-wc-stripe-order-handler.php

```diff
@@ -40,6 +42,77 @@ public static function get_instance() {
 		return self::$_this;
 	}
 
+	/**
+	 * Disables the ability to edit for orders with uncaptured payment.
+	 *
+	 * @param $editable boolean The current editability of the order.
+	 * @param $order WC_Order The order object.
+	 * @return boolean false if the order has uncaptured payment, true otherwise.
+	 */
+	public function disable_edit_for_uncaptured_orders( $editable, $order ) {
+		try {
+			$intent = $this->get_intent_from_order( $order );
+
+			if ( $intent && 'requires_capture' === $intent->status ) {
+				$editable = false;
+
+				// add hooks to change text about the reason when order items cannot be edited
+				add_action( 'woocommerce_admin_order_totals_after_total', [ $this, 'maybe_attach_gettext_callback' ] );
+				add_action( 'woocommerce_order_item_add_action_buttons', [ $this, 'maybe_unattach_gettext_callback' ] );
+			}
+		} catch ( Exception $e ) {
+			WC_Stripe_Logger::log( 'Error getting intent from order: ' . $e->getMessage() );
+		}
+
+		return $editable;
+	}
+
+	/**
+	 * Only attach the gettext callback when on admin edit order screen
+	 */
+	public static function maybe_attach_gettext_callback() {
```

Comment:
Followed [this](https://github.com/Automattic/woocommerce-subscriptions-core/blob/trunk/includes/admin/class-wc-subscriptions-admin.php#L1848-L1900).